department-of-veterans-affairs/abd-vro
# MCP-1479 Rhinitis condition processor

## What was the problem?

There was no support for a claim of a rhinitis condition.

Associated tickets or Slack threads:

- [MCP-1479](https://amida.atlassian.net/browse/MCP-1479)

## How does this fix it?[^1]

The processor will assess a patients record and include any of the following diagnostic codes in the evidence summary, if a related condition is found:
6522 Allergic or vasomotor rhinitis
6523 Bacterial rhinitis
6524 Granulomatous rhinitis

For right now the diagnostic code 6522 **must** be used in the full-health-assessment endpoint to reach the python processor queue (related ticket to map multiple codes to a single python processor https://amida.atlassian.net/browse/MCP-2053).

Medications and procedures are sorted by date. Medications are flagged for relevance to rhinitis and a suggested category is provided. Procedures are included only if they match identified procedures for rhinitis. 



## How to test this PR

 -   ./gradlew -PenablePrototype=true build check docker
 -   export COMPOSE_PROFILES=prototype
 -  ./gradlew :app:dockerComposeDown :app:dcPrune :app:dockerComposeUp

- Step 3 Try the full process endpoint with `diagnosticCode` 6522 and `ICN` 9000682

[^secrel]


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
